src/builtin.js

```diff
@@ -1061,7 +1061,12 @@ Sk.builtin.next_ = function next_(iter, default_) {
         if (default_) {
             return default_;
         }
-        throw new Sk.builtin.StopIteration();
+        const v = iter.$value;
+        if (v !== undefined && v !== Sk.builtin.none.none$) {
```

Comment:
Yes - you'd think this was redundant - but annoyingly - since we've lost the original StopIteration we don't know if was 
1. `StopIteration()`
2. `StopIteration(None)`

And if you return from a generator it's always `StopIteration()` over `StopIteration(None)`

In these two cases the `.value` property is both None. But the `.args` are different in these cases.
Which makes the reprs different
Which makes the tests fail

I've added a comment about this in the latest pr.
Essentially this is making a best guess that it was probably `StopIteration()` over `StopIteration(None)`
🤷

Reply:
I think I've fixed this redundancy #1306

Reply:
🤯